pyrefinebio/script.py

```diff
@@ -0,0 +1,126 @@
+    name = "dict"
+
+    def convert(self, value, param, ctx):
+        try:
+            return json.loads(value)
+        except:
+            self.fail(
+                "expected valid string in a dict form, "
+                "got: {0}".format(value),
+                param,
+                ctx
+            )
+
+
+class ListParamType(click.ParamType):
+    name = "list"
+
+    def convert(self, value, param, ctx):
+        try:
+            return value.split()
+        except:
+            self.fail(
+                "expected valid string in list form, "
+                "got: {0}".format(value),
+                param,
+                ctx
+            )
+
+
+@cli.command()
+@click.option("--entity", default="", help="entity to print help for")
+def help(entity=None):
+    """
+    Prints out information about pyrefinebio's classes and functions.
+    To get information about a class method, pass in the class name and method name  
+    separated by either a space or a `.`
+
+    Example:
+        $ refinebio help --entity "Sample.search"
+    """
+    hlf.help(entity)
+
+
+@cli.command()
+@click.option("--path", required=True, help="Path that the dataset should be downloaded to")
+@click.option("--email-address", required=True, help="The email that will be contacted with info related to the Dataset")
+@click.option("--dataset-dict", default=None, type=DictParamType(), help="A fully formed Dataset `data` attribute. Use this parameter if you want to specify specific Samples for your Dataset. Ex: '{\"GSE74410\": [\"GSM1919903\"]}' ")
+@click.option("--experiments", default=None, type=ListParamType(), help="A space separated list of experiment accession codes. Ex: 'SRP051449 GSE44421 GSE44422'")
+@click.option("--aggregation", default="EXPERIMENT", type=click.Choice(("EXPERIMENT", "SPECIES", "ALL"), case_sensitive=True), help="How the Dataset should be aggregated")
+@click.option("--transformation", default="NONE", type=click.Choice(("NONE", "MINMAX", "STANDARD"), case_sensitive=True), help="The transformation for the Dataset")
+@click.option("--skip-quantile-normalization", default=False, help="Control whether the Dataset should skip quantile normalization for RNA-seq Samples")
+def download_dataset(
+    path,
+    email_address,
+    dataset_dict,
+    experiments,
+    aggregation,
+    transformation,
+    skip_quantile_normalization
+):
+    """
+    Automatically constructs a Dataset, processes it, waits for it
+    to finish processing, then downloads it to the path specified.
+    """
+    try:
+        hlf.download_dataset(
+            path,
+            email_address,
+            dataset_dict,
+            experiments,
+            aggregation,
+            transformation,
+            skip_quantile_normalization
+        )
+    except DownloadError as e:
+        raise click.ClickException(e.message)
+
+
+@cli.command()
+@click.option("--path", help="Path that the Compendium should be downloaded to")
```

Comment:
Could this default to the current directory? I guess it'd need to wait on https://github.com/AlexsLemonade/refinebio-py/issues/31

Reply:
I added a comment there.

Reply:
I didn't mean to wait to re-review this long, but since we have a PR open for #31 maybe we can just wait until it lands to merge this so we can make this change here?

Reply:
added default path!